### PR TITLE
[luci] Introduce partition IRs

### PR DESCRIPTION
**Pull request by seanshpark**
This will introduce luci partition IRs for node and graph.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #6230, from draft #6231

File: compiler/luci/partition/src/PartitionIR.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __LUCI_PARTITON_IR_H__
+#define __LUCI_PARTITON_IR_H__
```
Comment (mhs4670go) [inline compiler/luci/partition/src/PartitionIR.h:18]:
```suggestion
#ifndef __LUCI_PARTITION_IR_H__
#define __LUCI_PARTITION_IR_H__
```

File: compiler/luci/partition/src/PartitionIR.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+#include <vector>
+
+namespace luci
+{
+
+struct PGraph;
+
+/**
+ * @brief Partition Node with CircleNode with group name
+ * @note  node just points to source luci::CircleNode * NOT the cloned node
```
Comment (llFreetimell) [inline compiler/luci/partition/src/PartitionIR.h:34]:
```suggestion
 * @note  node just points to source luci::CircleNode, NOT the cloned node
```

**Comment by seanshpark** [comment]
@nnfw-bot test nncc-debug

**Comment by seanshpark** [comment]
@nnfw-bot test nncc-debug

**Comment by seanshpark** [comment]
@nnfw-bot test nncc-debug

**Review by llFreetimell — APPROVED** [review]
LGTM

**Review by mhs4670go — APPROVED** [review]
LGTM

**Pull request merged**